src/evaluation/stats.jl

```diff
@@ -0,0 +1,142 @@
+using DrWatson
+using BSON
+using FileIO
+using Random
+using DataFrames
+using Statistics
+using StatsBase
+using EvalMetrics
+
+# metric names and settings 
+const BASE_METRICS = ["auc", "auprc", "tpr_5", "f1_5"]
+const PAT_METRICS = ["pat_1", "pat_5", "pat_10", "pat_20"]
+
+"""
+	_prefix_symbol(prefix, s)
+
+Modifies symbol `s` by adding `prefix` with underscore.
+"""
+_prefix_symbol(prefix, s) = Symbol("$(prefix)_$(s)")
+
+
+"""
+	_precision_at(p, scores, labels)
+
+Computes precision on portion `p` samples with highest score.
+"""
+function _precision_at(p, labels, scores)
+	pN = Int(round(floor(p*length(labels)))) # round is more forgiving
+	if pN > 0
+		sp = sortperm(scores, rev=true)[1:pN]
+		# @info sp scores[sp] labels[sp]
+		return EvalMetrics.precision(labels[sp], ones(eltype(labels), pN))
+	else
+		return NaN
+	end
+end
+
+"""
+	compute_stats(f::String)
+
+Computes evaluation metrics from the results of experiment in serialized bson at path `f`.
+Returns a DataFrame row with metrics and additional metadata for groupby's.
+Hash of the model parameters is precomputed in order to make the groupby easier.
+"""
+function compute_stats(f::String)
+	r = load(f)
+	row = (
+		modelname = r[:modelname],
+		dataset = r[:dataset],
+		phash = hash(r[:parameters]),
+		parameters = savename(r[:parameters], digits=6), 
+		seed = r[:seed])
+	
+	if Symbol("anomaly_class") in keys(r)
+		row = merge(row, (anomaly_class = r[:anomaly_class],))
+	end
+
+	for splt in ["val", "tst"]
+		scores = r[_prefix_symbol(splt, :scores)]
+		labels = r[_prefix_symbol(splt, :labels)]
+
+		if length(scores) > 1
+			invalid = isnan.(scores)
+			ninvalid = sum(invalid)
+
+			if ninvalid > 0
+				invrat = ninvalid/length(scores)
+				invlab = labels[invalid]
+				cml = countmap(invlab)
```

Comment:
did not know `countmap`, thats useful

Reply:
Right!? In python I so got to used to pandas's value counts (which worked only on their series) and was really glad that they added something like that to Julia.